bcf_sr_sort.c

```diff
@@ -24,21 +24,30 @@
 
 #include "bcf_sr_sort.h"
 #include "htslib/khash_str2int.h"
+#include <errno.h>
 
 #define SR_REF   1
 #define SR_SNP   2
 #define SR_INDEL 4
 #define SR_OTHER 8
 #define SR_SCORE(srt,a,b) (srt)->score[((a)<<4)|(b)]
 
+static inline void *die_if_null(void *ptr)
+{
+    if ( ptr ) return ptr;
+    errno = ENOMEM;
+    fprintf(stderr, "[E::%s] %s\n", __func__,strerror(errno));
+    exit(1);
```

Comment:
I suspect this isn't really what Rob expected.

It's OK for an application to do this, but when using a library I'd never expect it to simply call ```exit``` on me.  Instead the functions calling malloc should return an error indicator (eg the traditional 0 vs -1 or pointer vs NULL) if they fail so the calling application can do whatever is appropriate.

Normally that will also be exit, but perhaps after shutting things down, tidying up temporary files and similar.